ext/mysqlnd/mysqlnd_loaddata.c

```diff
@@ -156,6 +156,8 @@ mysqlnd_handle_local_infile(MYSQLND_CONN_DATA * conn, const char * const filenam
 
 	if (!(conn->options->flags & CLIENT_LOCAL_FILES)) {
 		php_error_docref(NULL, E_WARNING, "LOAD DATA LOCAL INFILE forbidden");
+		SET_CLIENT_ERROR(conn->error_info, CR_NOT_IMPLEMENTED, UNKNOWN_SQLSTATE,
```

Comment:
CR_NOT_IMPLEMENTED doesn't really seem fitting here, as the operation exists, it just not permitted. In the absence of something more specific, I'd use CR_UNKNOWN_ERROR.

Reply:
Wouldn't you agree that by configuring the client in a certain way, the requested action is no longer possible? If the server would disallow `LOAD FILE`, not-implemented might be the correct code, right?
I wouldn't mind changing it, there isn't a perfect error-code unfortunately

Reply:
It's the difference between "Not Found" and "Forbidden". Yes, in both cases the resource is unavailable, but the reason is very different.

On the server side ER_NOT_ALLOWED_COMMAND is used for this, but there doesn't seem to be a client equivalent.